src/main/groovy/com/morpheusdata/commvault/CommvaultBackupExecutionProvider.groovy

```diff
@@ -207,4 +223,9 @@ class CommvaultBackupExecutionProvider implements BackupExecutionProvider {
 		return ServiceResponse.success()
 	}
 
+	private logoutSession(authConfig) {
+		authConfig.token = authConfig.token ?: CommvaultBackupUtility.getToken(authConfig.apiUrl, authConfig.username, authConfig.password)?.token
```

Comment:
This method should not fetch a token. If `authConfig` has a token, then logout. If `authConfig` doesn't have a token then we have no session to logout.